feeluown/containers/player_control_panel.py

```diff
@@ -27,9 +27,61 @@ class SongBriefLabel(QLabel):
 
     def __init__(self, app):
         super().__init__(text=self.default_text, parent=None)
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Preferred)
         self._app = app
+        self.t = QTimer()
+        self.txt = self.raw_text = '...'
+        self.font_metrics = QFontMetrics(QApplication.font())
+        self.text_rect = self.font_metrics.boundingRect(self.raw_text)
+        self.newX = 0
+
+        self.t.timeout.connect(self.change_text_position)
         self._fetching_artists = False
 
+    def change_text_position(self):
+        if not self.parent().isVisible():
+            self.t.stop()
+            self.newX = 0
+            return
+        if self.text_rect.width() + self.newX > 0:
+            # control the speed of rolling
+            self.newX -= 5
+        else:
+            self.newX = self.width()
+        self.update()
+
+    def setText(self, text):
+        self.txt = self.raw_text = text
+        self.text_rect = self.font_metrics.boundingRect(self.raw_text)
+        self.newX = 0
+        self.update()
+
+    def enterEvent(self, event):
+        if self.txt != self.raw_text:
+            # decrease to make rolling more fluent
+            self.t.start(150)
+
+    def leaveEvent(self, event):
+        self.t.stop()
+        self.newX = 0
+        self.update()
+
+    def paintEvent(self, event):
+        painter = QPainter(self)
+        painter.setFont(QApplication.font())
+        painter.setPen(QColorConstants.Black)
```

Comment:
I think we should use systemc text color here.

For example, when the app is in dark theme, the text color should be white.

Reply:
So I make it fetch color from palette now :)